### DISCUSSION TITLE
How to correctly publish an open-source model, incorporating MESA

### DISCUSSION THREAD

**Discussion by SamuelGoodday** [mesa/mesa#1027] [Q&A] [answered]
Dear all,

I have enjoyed your good work on MESA during my PhD. Now, I want to publish an electricity market model, which also uses MESA. I am planning to publish under GPLv3 licence. I read that this is compatible with the Apache 2 licence of MESA. However, I want to be sure that I correctly refer to MESA and its contributers. 
https://wiki.openmod-initiative.org/wiki/Choosing_a_license
Therefore my questions is: Is this approach in line with your intentions?
Thank you very much in advance.

**Comment by tpike3** [comment]
I would say yes it is as your PhD adds knowledge so it makes sense to use a GPLv3 from Mesa's Apache. However, @jackiekazil is more knowledgeable than I on this subject.

I would ask that you point us to the final model. We would like to start maintaining a wiki on this repo to point to models/repos using mesa. Thanks!

**Reply by SamuelGoodday** [reply]
thanks for your reply.
The model can be found here : https://github.com/AncillaryServicesAcquisitionModel/ASAM

**Reply by tpike3** [reply]
@SamuelGoodday That is very cool thanks! I linked it on our Mesa Examples page.

**Reply by jackiekazil** [reply]
Ty for sharing!

**Comment by rht** [comment, marked as answer]
As long as you don't vendor actual Mesa code in your repo, I think you are fine.

If you do want to vendor Mesa, you can still use GPLv3 and additionally say that you vendor in Mesa's code with an Apache2 license)

Btw, I saw the citations in your paper (https://arxiv.org/pdf/2104.13047.pdf), you wrote 
> Project Mesa Team, MESA- Website (2016). URL https://mesa.readthedocs.io/en/master/overview.html

Shouldn't it be the Mesa paper instead?
Maybe we should add in the README.md on how to cite Mesa properly.

**Reply by rht** [reply]
@jackiekazil @dmasad is this the correct paper to cite from? https://link.springer.com/chapter/10.1007/978-3-030-61255-9_30

**Reply by SamuelGoodday** [reply]
Thanks for your reply. Indeed, a suggestion on how to cite MESA would be great. I will change it in my references as soon as you have added it to the readme.

**Reply by jackiekazil** [reply]
@rht that is the correct paper. Ty for pointing people to that!

**Reply by jackiekazil** [reply]
... should probably add that to the readme or something

**Reply by rht** [reply]
@jackiekazil I made #1046 for the citation info in the repo README.
